roles/nginx/tasks/main.yml

```diff
@@ -0,0 +1,20 @@
+---
+- name: install nginx
+  apt:
+    name: nginx
+    state: latest
+  notify:
+    - start nginx
+
+- name: enable nginx in ufw
+  ufw:
+    rule: allow
+    name: 'Nginx Full'
+  notify:
+    - enable ufw
+
+- include: ssl.yml
+  when: ssl
+
+- include: http.yml
```

Comment:
Kind of a hack, I actually include the http.yml either way, but just in different orders in case something doesn't need SSL (local usage, for example)